code/game/objects/structures/trash_pile.dm

```diff
@@ -0,0 +1,204 @@
+/obj/structure/trash_pile
+	name = "trash pile"
+	desc = "A heap of garbage, but maybe there's something interesting inside?"
+	icon = 'icons/obj/trash_piles.dmi'
+	icon_state = "randompile"
+	density = TRUE
+	anchored = TRUE
+
+	var/list/searched_by // Characters that have searched this trashpile, with values of searched time.
+	var/mob/living/hider		// Someone or something hiding inside our pile
+
+	var/chance_alpha	= 79	// Alpha list is junk items and normal random stuff.
+	var/chance_beta		= 20	// Beta list is actually maybe some useful illegal items. If it's not alpha or gamma, it's beta.
+	var/chance_gamma	= 1		// Gamma list is unique items only, and will only spawn one of each. This is a sub-chance of beta chance.
+
+	//These are types that can only spawn once, and then will be removed from this list.
+	//Alpha and beta lists are in their respective procs.
+	var/list/unique_gamma = list(
+		/obj/item/hand_tele,
+		/obj/item/rfd/construction,
+		/obj/item/reagent_containers/hypospray/cmo,
+		/obj/item/gun/projectile/improvised_handgun/loaded,
+		/obj/item/gun/launcher/crossbow,
+		/obj/item/gun/launcher/pneumatic
+		)
+
+/obj/structure/trash_pile/Initialize()
+	. = ..()
+	icon_state = pick("pile1","pile2","pilechair","piletable","pilevending","brtrashpile","microwavepile","rackpile","boxfort","trashbag","brokecomp")
```

Comment:
```suggestion
	if(icon_state == initial(icon_state))
		icon_state = pick(icon_states(icon) - icon_state)
```

Reply:
Incase if icon state is set to something specific on the map it stays that way.
Also future proofing if more icon states are added or changed.